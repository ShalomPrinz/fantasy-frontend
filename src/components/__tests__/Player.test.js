import React from 'react';
import renderer from 'react-test-renderer';
import { Player } from '../';

let tree = null;

beforeEach(() => {
    tree = null
})

const getJSON = (Component) => renderer.create(Component).toJSON()

describe('render', () => {
    describe('without props', () => {
        it('should not render anything', () => {
            tree = getJSON(<Player />);
        
            expect(tree).toMatchSnapshot();
        })
    })

    describe('partial props', () => {
        it('should not render anything when not getting widthUnits', () => {
            tree = getJSON(<Player width={10} />);
        
            expect(tree).toMatchSnapshot();
        })
        
        
        it('should not render anything when not getting width', () => {
            tree = getJSON(<Player widthUnits="px" />);
        
            expect(tree).toMatchSnapshot();
        })
        
        
        it('should render Player with default Jersey and without name label', () => {
            tree = getJSON(<Player width={10} widthUnits="px" />);
        
            expect(tree).toMatchSnapshot();
        })
    })

    describe('with props', () => {
        it ('should render full Player component', () => {
            tree = getJSON(<Player name="Shalom" team="Dortmund" width={10} widthUnits="px" />);
    
            expect(tree).toMatchSnapshot();
        })
    })
})
