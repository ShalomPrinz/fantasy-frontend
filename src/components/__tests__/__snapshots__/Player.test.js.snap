// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render partial props should not render anything when not getting width 1`] = `null`;

exports[`render partial props should not render anything when not getting widthUnits 1`] = `null`;

exports[`render partial props should render Player with default Jersey and without name label 1`] = `
<div
  className="text-center"
  style={
    Object {
      "width": "14px",
    }
  }
>
  <img
    className=""
    src="Barcelona.png"
    style={
      Object {
        "width": 10,
      }
    }
    title="Barcelona"
  />
</div>
`;

exports[`render with props should render full Player component 1`] = `
<div
  className="text-center"
  style={
    Object {
      "width": "14px",
    }
  }
>
  <img
    className=""
    src="Dortmund.png"
    style={
      Object {
        "width": 10,
      }
    }
    title="Dortmund"
  />
  <div
    className="text-truncate bg-default rounded p-1 w-100"
  >
    Shalom
  </div>
</div>
`;

exports[`render without props should not render anything 1`] = `null`;
